ui/components/PageStatus.tsx

```diff
@@ -17,13 +17,21 @@ function PageStatus({ conditions, suspended, className }: StatusProps) {
   const ok = suspended ? false : computeReady(conditions);
   const msg = suspended ? "Suspended" : computeMessage(conditions);
 
-  let iconType;
+  let iconType: IconType;
   if (suspended) iconType = IconType.SuspendedIcon;
-  else if (ok)
-    ok === ReadyType.Reconciling
-      ? (iconType = IconType.ReconcileIcon)
-      : (iconType = IconType.CheckCircleIcon);
-  else iconType = IconType.FailedIcon;
+  else {
+    switch (ok) {
+      case ReadyType.Reconciling:
+        iconType = IconType.ReconcileIcon;
+        break;
+      case ReadyType.Ready:
+        iconType = IconType.CheckCircleIcon;
+        break;
+      case ReadyType.NotReady:
+        iconType = IconType.FailedIcon;
+        break;
+    }
```

Comment:
Usually, `switch` should have the default case to make sure that all possible cases are covered.

Reply:
Oh. Usually in typed languages, a switch without a default against an enum means the compiler will make sure your matches are exhaustive - I assumed typescript was the same, but it seems I need type level trickery.

Reply:
Yes, just checked if TypeScript catches it: left only two specific cases without a default case, but nothing was reported by the editor or by lint.